### PR TITLE
Smoother roundstart take II

### PR DESCRIPTION
**Pull request by Cyberboss**
Reverts tgstation/tgstation#24368

Includes fixes from #24370 and #24364 

:cl: Cyberboss
experiment: Roundstart should now be a smoother experience... again
/:cl:

### CODE REVIEW

File: code/modules/mob/transform_procs.dm [hunk 1]
```diff
@@ -334,13 +337,27 @@
...
 
 //human -> robot
-/mob/living/carbon/human/proc/Robotize(delete_items = 0)
+/mob/proc/Robotize(delete_items = 0)
+
+/mob/new_player/Robotize(delete_items = 0, transfer_after = TRUE)
+	spawning = 1
+	if(new_character)
+		new_character = new_character.Robotize(FALSE, transfer_after)
+		testing("New new_character is now [new_character]([new_character.type])")
```
Comment (coiax) [inline code/modules/mob/transform_procs.dm]: Is this sentence correct? "New new_character"?

Comment (Cyberboss) [inline code/modules/mob/transform_procs.dm]: `new_character` is the var name, so yeah

Comment (Cyberboss) [inline code/modules/mob/transform_procs.dm]: No, but I'm changing this 1 sec

**Comment by Cyberboss** [comment]
The fact that the silicons were hella snowflakey hurts my soul

**Comment by Cyberboss** [comment]
Doesn't look too bad, people can join while the game is starting and the ai is being shot by its turrets. 9/10 for me

**Comment by Cyberboss** [comment]
Ready

File: code/modules/jobs/job_types/job.dm [hunk 2]
```diff
@@ -44,15 +44,17 @@
 	var/outfit = null
 
 //Only override this proc
-/datum/job/proc/after_spawn(mob/living/carbon/human/H)
+//H is usually a human unless an /equip override transformed it
+/datum/job/proc/after_spawn(mob/living/H, mob/M)
+	//do actions on H but send messages to M as the key may not have been transferred_yet
```
Comment (RemieRichards) [inline code/modules/jobs/job_types/job.dm]: :(

Comment (Cyberboss) [inline code/modules/jobs/job_types/job.dm]: This was always the case though, I just updated the documentation

Comment (Cyberboss) [inline code/modules/jobs/job_types/job.dm]: As for `mob/M` a necessary evil

Comment (RemieRichards) [inline code/modules/jobs/job_types/job.dm]: What, so these messages just never sent then?

**Comment by Cyberboss** [comment]
Borgs not syncing to AIs

**Comment by Cyberboss** [comment]
Ready

File: code/modules/jobs/job_types/job.dm (hunk 2 above)
Comment (AnturK) [inline code/modules/jobs/job_types/job.dm]: I don't know if it worked but i like the added explicit link between to be spawned character.

**Pull request merged**